src/components/structured-list/structured-list.js

```diff
@@ -0,0 +1,86 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventMatches from '../../globals/js/misc/event-matches';
+
+class StructuredList extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * StructuredList
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The root element of tables
+   * @param {Object} [options] the... options
+   * @param {string} [options.selectorInit] selector initialization
+   * @param {string} [options.selectorRow] css selector for selected row
+   */
+  constructor(element, options) {
+    super(element, options);
+
+    this.element.addEventListener('keydown', (evt) => {
+      this._arrowKeydown(evt);
+      this._selectKeydown(evt);
+    });
+  }
+
+  _selectKeydown(evt) {
+    const selectedRow = eventMatches(evt, this.options.selectorRow);
+    if (selectedRow) {
+      const input = this.element.querySelector(
+        `#${selectedRow.getAttribute('for')}.bx--structured-list-input`,
+      );
+      if (evt.which === 13) {
+        input.checked = true;
+      }
+
+      if (evt.which === 32) {
+        evt.preventDefault();
+        input.checked = true;
+      }
+    }
+  }
+
+  // _getInput(item) {
+  //   const id = `#${item.getAttribute('for')}`;
+  //   return this.element.querySelector(`${id}.bx--structured-list-input`);
+  // }
+
+  _arrowKeydown(evt) {
+    const selectedRow = eventMatches(evt, this.options.selectorRow);
```

Comment:
> Hi @hellobrian seems that <input class=“bx—structured-list-input>, that is outside the row element, receives up/down arrow key events. eventMatches() merely searches for an element matching the given selector up in DOM tree, beginning with the event target. That said, you need to associate with such <input> to the table row to get the right row index.

I think what you were saying was that `eventMatches` would return `undefined` because eventually, `evt.target` returns `<input class=“bx—structured-list-input>` and not the associated `<label>`. For some reason, this is only happening in Firefox.